tools/ci-build/publisher/src/subcommand/fix_manifests.rs

```diff
@@ -210,6 +224,24 @@ fn update_dep(table: &mut Table, dep_name: &str, versions: &VersionView) -> Resu
     }
 }
 
+// Convert `package_version` into a tilde version requirement
+//
+// For instance, given `package_version` like `0.12.3`, the function returns `~0.12`.
+// The fact that this function takes a `semver::Version` means one can only convert a complete
+// semver `x.y.z` into a tilde version requirement, but not those like `0.21.0-alpha.1`.
+fn convert_to_tilde_requirement(package_version: &semver::Version) -> Result<String> {
+    // `package_version` is from the `semver` crate which requires versions to have 3 components,
+    // major, minor, and patch. So it is safe to assume its string value follows that format.
+    let package_version = package_version.to_string();
+    let package_version = match package_version.rfind('.') {
```

Comment:
Consider adding an `assert!` that makes sure there are only two `.` characters so that it breaks instead of doing something weird for versions like `0.21.0-alpha.1`

Reply:
True, a semver can contain more than two dots. Added assert in 137822d.